src/approve.ts

```diff
@@ -30,43 +30,43 @@ export async function approve(
     core.info(`Current user is ${login}`);
 
     core.info(`Getting pull request #${prNumber} info`);
-    const pull_request = await client.rest.pulls.get({
-      owner: context.repo.owner,
-      repo: context.repo.repo,
-      pull_number: prNumber,
-    });
-    const commit = pull_request.data.head.sha;
+    const { owner, repo } = context.repo;
+    const queryParams = { owner, repo, pull_number: prNumber };
+    const [pullRequest, reviews, requestedReviewers] = await Promise.all([
```

Comment:
Hi @hmarr, while doing the fixes, I realized we can get `requested_reviewers` from `pull_request`, I was about to PR this...

```javascript
for (const review of reviews.data) {
      if (
        review.user?.login == login &&
        review.commit_id == commit &&
        review.state == "APPROVED" &&
        pull_request.data
        .requested_reviewers?.filter(
          reviewer => reviewer.login == login
        ).length == 0
      ) {
        core.info(
          `Current user already approved pull request #${prNumber}, nothing to do`
        );
        return;
      }
    }
```

Reply:
@hmarr nevermind I PRd https://github.com/hmarr/auto-approve-action/pull/210, let me know your thoughts...

Reply:
Great, I'll close this out in favour of your pull request. Thanks for contributing!